service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -101,6 +101,8 @@ private[lagom] object Producer {
     def generalHandler: Receive = {
       case Failure(e) =>
         throw e
+      case t: Throwable =>
+        throw t
```

Comment:
The stream connecting the akka persistence query, the user-land code and the broker producer is materialized to a `Future[Done]` that is [piped](https://github.com/lagom/lagom/blob/1af0f3d1f07c6d9d7a77585ebf2b72962d520f3e/service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala#L137-L137) to `self`. 

When there's an exception in either part of the stream, the exception propagates and the `Future[Done]` is completed with a `Failure`. when piping that to `self` the message received is the exception not the Failure.